var/spack/repos/builtin/packages/libxml2/package.py

```diff
@@ -42,8 +43,11 @@ class Libxml2(Package):
 
     def install(self, spec, prefix):
         if '+python' in spec:
-            site_packages_dir = os.path.join(prefix, 'lib/python%s.%s/site-packages' %(spec['python'].version[:2]))
-            python_args = ["--with-python=%s" % spec['python'].prefix, "--with-python-install-dir=%s" % site_packages_dir]
+            site_packages_dir = os.path.join(prefix,
+                                             'lib/python%s/site-packages'
+                                             % (spec['python'].version.up_to(2)))
```

Comment:
Hmm, I see pros and cons to each way. I kind of like the module-scoped variable for this and the lib/include dirs. I am not sure of the history of this package and why the variable was not used. When I looked at it I was focused on the string formatting so did not think about it.

In the end I think it is a matter of learning the spack shortcuts that are provided for things like this.